output.R

```diff
@@ -161,36 +65,37 @@ choose_filename_prefix <- function(modules, title = "") {
 }
 
 if (exists("source_include")) {
-  comp <- FALSE
-} else if (!exists("comp")) {
-  comp <- choose_mode("Please choose the output mode")
+  comp <- "single"
+} else if (! exists("comp")) {
+  modes <- c("single" = "Output for single run", "comparison" = "Comparison across runs", "export" = "Export", "exit" = "Exit")
+  comp <- names(modes)[which(chooseFromList(unname(modes), type = "output mode", multiple = FALSE, returnBoolean = TRUE))]
+  if (comp == "exit") q()
 }
+if (isFALSE(comp)) comp <- "single" # legacy from times only two comp modes existed
 if (isTRUE(comp)) comp <- "comparison"
 
-if (comp == "Exit") {
-  q()
-} else if (comp == "comparison" | comp == "export") {
-  print("comparison")
-  # Select output modules if not defined by readArgs
-  if (!exists("output")) {
-    output <- choose_module(paste0("./scripts/output/", comp),
-                            "Please choose the output module to be used for output generation")
-  }
+if (! exists("output")) {
+  modules <- gsub("\\.R$", "", grep("\\.R$", list.files(paste0("./scripts/output/", if (isFALSE(comp)) "single" else comp)), value = TRUE))
+  output <- chooseFromList(modules, type = "modules to be used for output generation", addAllPattern = FALSE)
+}
+
+if (comp %in% c("comparison", "export")) {
   # Select output directories if not defined by readArgs
-  if (!exists("outputdir")) {
+  if (! exists("outputdir")) {
     if ("policyCosts" %in% output) {
       message("\nFor policyCosts, specify policy runs and reference runs alternatingly:")
       message("3,1,4,1 compares runs 3 and 4 with 1.")
     }
-    if (!exists("remind_dir")) {
-      temp <- choose_folder("./output", "Please choose the runs to be used for output generation")
-      outputdirs <- temp
-      for (i in seq_along(temp)) {
-        outputdirs[i] <- file.path("output", temp[i])
-      }
-    } else {
-      temp <- choose_folder(remind_dir, "Please choose the runs to be used for output generation, separate with comma")
-      outputdirs <- temp
+    dir_folder <- if (exists("remind_dir")) remind_dir else "./output"
+    dirs <- basename(dirname(Sys.glob(file.path(dir_folder, "*", "fulldata.gdx"))))
+    # DK: The following outcommented lines are specially made for listing results of coupled runs
+    # runs <- findCoupledruns(folder)
+    # dirs <- findIterations(runs,modelpath=folder,latest=TRUE)
+    # dirs <- sub("./output/","",dirs)
+    temp <- chooseFromList(dirs, type = "runs to be used for output generation", returnBoolean = FALSE,
+                                      multiple = TRUE)
+    outputdirs <- file.path("output", temp)
```

Comment:
Please use a better name than `temp`. I know you didn't invent it, but it gets even harder to understand what `temp` is now that the function call is a generic `chooseFromList` instead of `choose_folder`. Maybe `folders` would be a good name instead of `temp`?

Reply:
or `selected_folders`

Reply:
I used selectedDirs

Reply:
ah, right, it seems I will never use camelCase on first try, always get caught by the linter. :rofl: